docs/DeveloperGuide.md

```diff
@@ -1,34 +1,346 @@
+    - Quiz
+        * Quiz
+        * Result
+        * ResultList
+        * TopicQuiz
+        * SubjectQuiz
+    - Flashcard
+    - Subject
+    - SubjectList
+    - Topic
+    - TopicList
+    
+
+* Command: In charge of reading the command and calling the relevant methods. The package itself holds
+the following packages.
+    - Flashcard
+    - Subject
+    - Topic
+    - Task
+    
+* Exceptions: Deals with the various exceptions which are thrown in the application.
+    -flashcard
+    - storage
+    - subject
+    - task
+    - topic
+* Parser: Parses the commands and instantiates the relevant command classes.
+* Storage: Stores the data in an external folder
+* Task: Contains the classes for the different types of tasks.
+* Ui: In charge of interaction with the user.
+
+### Quiz package
+The `Storage` package holds the necessary classes for the quiz functionality of this application. The following are
+the classes in the package:an abstract`Quiz` class, a `Result` class, a `SubjectQuiz` class and a `TopicQuiz` class.
+
+#### Quiz class
+`Quiz` class is an abstract class in the `Quiz` package. It is holds the result of a quiz class and the list of flashcards
+from any subject or topic classes which the user calls the quiz for. Furthermore, it contains a checkAnswer() method the
+checks the answer that the user had given with the correct answer of the quiz. If the user enters the correct answer, the 
+existing score is incremented by one. Else, the contents of the flashcards and the incorrect answer provided by
+the user is transferred to the  `incorrectAnswers` list. Once the user finished the quiz, the application would print the 
+questions that the users did not answer correctly, along the the answer that was provided by the user. 
+
+#### SubjectQuiz
+`SubjectQuiz` class inherits from the `Quiz` class and initiates the quiz. The `startQuiz` method calls for the  `setupQuiz` method checks for the presence
+of topics or flashcards. Else, the application throws the `NoTopicException` for the former, and the 
+`NoFlashcardException` for the latter. If the topics have flashcards, then these are transferred to the 
+`SubjectQuiz` class, while the maximum score of the quiz is set to be the total number of flashcards in the list of flashcards
+present in the `SubjectQuiz` class . 
+
+The application now returns to the `startQuiz` method and the current score of the quiz is set to zero. Subsequently, the
+application begins printing the questions from the flashcards and checks the answer that you provide. If you want to stop the 
+quiz, use the `stop` command. The application will then print out the score that you obtain. If you complete the quiz
+then the application not only prints the score, but it also prints the incorrect answers from the quiz.
+
+The following diagram shows how you can initiate the quiz for a subject.
+![first](https://user-images.githubusercontent.com/46095141/97313097-3866e200-18a1-11eb-9525-73e38ceb7cbe.png)
```

Comment:
Should the diagrams be standardised (perhaps by using the same drawing software) to improve consistency?
![image](https://user-images.githubusercontent.com/22489162/97525989-24c39480-19e3-11eb-9c1c-15991bdd2fdd.png)
![image](https://user-images.githubusercontent.com/22489162/97526009-30af5680-19e3-11eb-8a6d-faef11f8d3f6.png)